cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -102,7 +103,10 @@ public void addPartition(PartitionKey key, String path) {
               Bytes.toBytes(entry.getValue().toString()));            // "<string rep. of value>"
     }
     partitions.put(put);
+    addPartitionToExplore(key, path);
```

Comment:
No. This is a common problem. We have a Jira open to make DDL operations transactional. Itis complex to do that and beyond the scope of this Jira. But I can add a TODO here.